### PR TITLE
[luci] Introduce Shape related for unit tests

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce Shape related types and methods for unit testing.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #6110, from draft #6111

**Review by llFreetimell — APPROVED** [review]
LGTM

File: compiler/luci/pass/src/test/TestShape.test.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "TestShape.h"
+
+/**
+ * @note This file does not hold any test cases but provides methods for tests
```
Comment (mhs4670go) [inline compiler/luci/pass/src/test/TestShape.test.cpp:20]: Then, is it okay to rename this file to `TestShape.cpp`?

**Review by jinevening — APPROVED** [review]
LGTM

File: compiler/luci/pass/src/test/TestShape.test.cpp (hunk 1 above)
Comment (seanshpark) [reply compiler/luci/pass/src/test/TestShape.test.cpp:20]: `TestShape.cpp` will be added as a source to luci library `luci_pass` that is not used anywhere else, so would be inappropriate.

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**